File: assertion/python/question4.py
from math import exp
import random as __random

_REPLACE_ME_WITH_DIRECTIVES_

_REPLACE_ME_WITH_SOLUTION_

def main():
    testCases = [
        {
            "arguments": "findHeaterPower(100212373)",
            "got": findHeaterPower(100212373),
            "expected": 19
        }
    ]

    def workingAnswer(n):
        s = str(n)
        sum = 0
        for i in range(len(s)):
            sum += int(s[i])
        return sum

    for _ in range(9):
        n = __random.randint(1000000000, 9999999999)
        expected = workingAnswer(n)
        got = findHeaterPower(n)
        arguments = f"findHeaterPower({n})"
        testCases.append({ "got": got, "expected": expected, "arguments": arguments })

    for i, test in enumerate(testCases):
        if type(test["got"]) != int and type(test["got"]) != float and type(test["got"]) != str:
            print(f"# {i+1} FAILED")
            print(f"> ARGUMENTS {test['arguments']}")
            print(f"> EXPECTED {test['expected']}")
            print(f"> GOT {test['got']}")
            continue

        if type(test["got"]) == str and test["got"].isdigit() == False:
            print(f"# {i+1} FAILED")
            print(f"> ARGUMENTS {test['arguments']}")
            print(f"> EXPECTED {test['expected']}")
            print(f"> GOT {test['got']}")
            continue

        if int(test["got"]) == int(test["expected"]):
            print(f"# {i+1} PASSING")
            print(f"> ARGUMENTS {test['arguments']}")
            print(f"> EXPECTED {test['expected']}")
            print(f"> GOT {test['got']}")
        else:
            print(f"# {i+1} FAILED")
            print(f"> ARGUMENTS {test['arguments']}")
            print(f"> EXPECTED {test['expected']}")
            print(f"> GOT {test['got']}")

if __name__ == "__main__":
    main()

File: assertion/php/question4.php
<?php

_REPLACE_ME_WITH_DIRECTIVES_

_REPLACE_ME_WITH_SOLUTION_

function main(): void {
    $test_cases = [
        [
            "arguments" => "findHeaterPower(100212373)",
            "got" => findHeaterPower(100212373),
            "expected" => 19
        ]
    ];

    function working_answer(int $n) {
        $s = strval($n);
        $sum = 0;
        for ($i = 0; $i < strlen($s); $i++) {
            $sum += intval($s[$i]);
        }

        return $sum;
    }

    for ($i = 0; $i < 9; $i++) {
        $n = rand(1000000000, 9999999999);
        $expected = working_answer($n);
        $got = findHeaterPower($n);
        $arguments = "findHeaterPower($n)";
        array_push($test_cases, [
            "got" => $got,
            "expected" => $expected,
            "arguments" => $arguments
        ]);
    }

    for ($i = 0; $i < count($test_cases); $i++) {
        $test = $test_cases[$i];

        if ($test["got"] === $test["expected"]) {
            echo "# ". $i + 1 . " PASSING\n"; 
            echo "> ARGUMENTS " . $test["arguments"] . "\n";
            echo "> EXPECTED " . $test["expected"] . "\n";
            echo "> GOT " . $test["got"] . "\n";
        } else {
            echo "# " . $i + 1 . " FAILED\n";
            echo "> ARGUMENTS " . $test["arguments"] . "\n";
            echo "> EXPECTED " . $test["expected"] . "\n";
            echo "> GOT " . $test["got"] . "\n";
        }
    }
}

main();